node/modules/storageminer_dagstore.go

```diff
@@ -23,88 +23,70 @@ const (
 )
 
 // NewMinerAPI creates a new MinerAPI adaptor for the dagstore mounts.
-func NewMinerAPI(lc fx.Lifecycle, r repo.LockedRepo, pieceStore dtypes.ProviderPieceStore, sa mdagstore.SectorAccessor) (mdagstore.MinerAPI, error) {
-	cfg, err := extractDAGStoreConfig(r)
-	if err != nil {
-		return nil, err
-	}
-
-	// caps the amount of concurrent calls to the storage, so that we don't
-	// spam it during heavy processes like bulk migration.
-	if v, ok := os.LookupEnv("LOTUS_DAGSTORE_MOUNT_CONCURRENCY"); ok {
-		concurrency, err := strconv.Atoi(v)
-		if err == nil {
-			cfg.MaxConcurrencyStorageCalls = concurrency
+func NewMinerAPI(cfg config.DAGStoreConfig) func(fx.Lifecycle, repo.LockedRepo, dtypes.ProviderPieceStore, mdagstore.SectorAccessor) (mdagstore.MinerAPI, error) {
```

Comment:
Unrelated?

Reply:
This gets rid of the `extractDAGStoreConfig` which casts the `config`, so it is somewhat related and necessary as part this work, but I agree slightly tangential to the rest of the changes.